src/QueryBuilder/QueryBuilder/helpers/valueBuilder.test.js

```diff
@@ -139,3 +139,29 @@ describe('valueBuilder', () => {
     expect(valueBuilder({ value, field, operator, fieldOptions })).toBe(getQuotedStr(value, true));
   });
 });
+
+describe('retainValueOnOperatorChange', () => {
+  it('should return the previous value when the operator type is the same', () => {
+    const result = retainValueOnOperatorChange(OPERATORS.EQUAL, OPERATORS.EQUAL, 'someValue');
+
+    expect(result).toBe('someValue');
+  });
+
+  it('should return the previous value when switching between comparison and comparison array', () => {
+    const result = retainValueOnOperatorChange(OPERATORS.EQUAL, OPERATORS.IN, 'someValue');
+
+    expect(result).toBe('someValue');
+  });
+
+  it('should return the empty value of the array when switching from comparison array to comparison', () => {
+    const result = retainValueOnOperatorChange(OPERATORS.IN, OPERATORS.EQUAL, [1, 2, 3]);
+
+    expect(result).toBe(1);
+  });
```

Comment:
Not "the empty value of the array" but "the first value of the array". At least, "first value of the array" is what's implemented up on valueBuilder.js line 102.

Reply:
Updated the text regarding your comment